### ISSUE TITLE
Delay Check for Parasitic Powering and Setting Resolution

### ISSUE THREAD

**Issue by gfvalvo** [milesburton/Arduino-Temperature-Control-Library#119]
Hi,
Regarding the ‘DallasTemperature::begin()’ method. I’m wondering if perhaps the following lines:

```
  if (validAddress(deviceAddress)) {
    if (!parasite && readPowerSupply(deviceAddress))
      parasite = true;

    bitResolution = max(bitResolution, getResolution(deviceAddress));

    devices++;
    if (validFamily(deviceAddress)) {
      ds18Count++;
    }
  }
```

Should be rearranged as:

```
  if (validAddress(deviceAddress)) {
    devices++;
    if (validFamily(deviceAddress)) {

      if (!parasite && readPowerSupply(deviceAddress))
        parasite = true;

      bitResolution = max(bitResolution, getResolution(deviceAddress));
      ds18Count++;
    }
  }
```

Thus, the check for parasitic powering and setting of the temperature resolution will be postponed until we’re sure that the currently located device is actually a member of the DS18... family. This could be important if other types of devices are also on the same OneWire bus.

**Comment by RobTillaart** [comment]
Seems valid to me. Can you make a PR for it?

**Comment by RobTillaart** [comment]
#150 should be merged first
